tools/perf/update_profile.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+# Copyright (c) 2023 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+""" A script to automate updating the perf profiles for perf CI
+
+It downloads and unpacks an old profile, run the browser twice to upload the
+components/other things, makes a new zip and uploads it to GCP.
+
+Example usage:
+vpython3 update_profile.py brave-typical-v1.46.52 brave-typical-mac-v1.55.98
+  "/Applications/Brave Browser Nightly.app/Contents/MacOS/Brave Browser Nightly"
+"""
+
+import argparse
+import json
+import logging
+import os
+import subprocess
+import sys
+import tempfile
+import time
+
+import components.path_util as path_util
+import components.perf_profile as perf_profile
+
+from lib.util import make_zip, scoped_cwd
+
+
+def EraseVariationsFromLocalState(local_state_path: str):
+  local_state = {}
+  with open(local_state_path, 'r', encoding='utf8') as f:
+    local_state = json.load(f)
+    for k in list(local_state.keys()):
+      if k.startswith('variation'):
+        local_state.pop(k, None)
+
+  with open(local_state_path, 'w', encoding='utf8') as f:
+    json.dump(local_state, f)
+
+
+def UpdateProfile(profile_name: str, browser_binary: str, new_profile_name: str,
+                  skip_upload: bool):
+  assert os.path.exists(browser_binary)
+  tmp = tempfile.mkdtemp(prefix='')
+  logging.info('Using temp directory %s', tmp)
+  profile_dir = perf_profile.GetProfilePath(profile_name, tmp)
+  logging.info('Profile unpacked in %s', profile_dir)
+
+  args = [
+      browser_binary,
+      f'--user-data-dir={profile_dir}',
+
+      # Enable perf related features to download the components
+      '--enable-brave-features-for-perf-testing',
+
+      # Speed up component updating
+      '--component-updater="fast-update"'
+  ]
+
+  # Run browser for 2 minutes to update the profile.
+  # Do this twice to clean the old components.
+  for _ in range(2):
+    logging.info('Run %s', ' '.join(args))
+    with subprocess.Popen(args) as process:
+      logging.info('Waiting for 2 minutes..')
+      time.sleep(120)
```

Comment:
i wonder if there is any diagnostic about whether all the components were actually downloaded

Reply:
@iefremov I believe we soon change that mechanism to use the recent components.